apiserver/facades/client/controller/controller.go

```diff
@@ -720,26 +721,57 @@ func (c *ControllerAPI) ModifyControllerAccess(args params.ModifyControllerAcces
 	hasPermission := err == nil
 
 	for i, arg := range args.Changes {
+		targetUserTag, err := names.ParseUserTag(arg.UserTag)
+		if err != nil {
+			result.Results[i].Error = apiservererrors.ServerError(errors.Annotate(err, "could not modify controller access"))
+			continue
+		}
```

Comment:
I wonder if they would be interested in seeing this failed login? It might be indicative of a rainbow attack.